### PR TITLE
feat: support iterator serialization

### PR DESCRIPTION
**Pull request by thiagowfx**
Bug: #562
Tested: Works for Map and Set iterators. Array and String iterators will need a separate change in V8.

### CODE REVIEW

**Review by OrKoN — APPROVED** [review]

File: src/bidiMapper/domains/script/realm.ts [hunk 1]
```diff
@@ -110,13 +110,16 @@ export class Realm {
...
-      cdpValue.result.type === 'object' &&
-      cdpValue.result.subtype === 'generator'
-    ) {
-      return {
-        type: 'generator',
-      };
+    if (cdpValue.result.type === 'object') {
+      switch (cdpValue.result.subtype) {
+        case 'generator':
+        case 'iterator':
```
Comment (Lightning00Blade) [inline src/bidiMapper/domains/script/realm.ts:116]: Nit: Add TODO about Array/String iterator

Comment (thiagowfx) [reply src/bidiMapper/domains/script/realm.ts:116]: This code is going to remain the same though. The TODO should be tracked in the bug (or in V8).

What I could do is add an unit test that is disabled and has a TODO.

Comment (thiagowfx) [reply src/bidiMapper/domains/script/realm.ts:116]: Added tests + TODO

File: src/bidiMapper/domains/script/realm.ts [hunk 2]
```diff
@@ -110,13 +110,16 @@ export class Realm {
...
     }
 
-    if (
-      cdpValue.result.type === 'object' &&
-      cdpValue.result.subtype === 'generator'
-    ) {
-      return {
-        type: 'generator',
-      };
+    if (cdpValue.result.type === 'object') {
```
Comment (sadym-chromium) [inline src/bidiMapper/domains/script/realm.ts:113]: This check is  done only for the top-level objects, but not for the nested one. Traversing nested data is done in [`#deepSerializedToBiDi`](https://github.com/GoogleChromeLabs/chromium-bidi/pull/1042/files#diff-d7af9adf42c7a4f4040462b9ff23cae43df9de6a355ee3197df6d9462485c1c3R93), but there is no `type` and `subtype` data.

**Pull request merged**